m4/odp_libconfig.m4

```diff
@@ -0,0 +1,36 @@
+# ODP_LIBCONFIG([ACTION-IF-FOUND], [ACTION-IF-NOT-FOUND])
+# -----------------------------------------------------
+AC_DEFUN([ODP_LIBCONFIG],
+[dnl
+##########################################################################
+# Set optional libconfig path
+##########################################################################
+AC_ARG_WITH([libconfig-path],
+[AS_HELP_STRING([--with-libconfig-path=DIR],
+[path to libconfig libs and headers (use system path if not provided)])],
+[export PKG_CONFIG_PATH="$PKG_CONFIG_PATH:$withval/lib/pkgconfig"],
+)
+
+##########################################################################
+# Check for libconfig availability
+##########################################################################
+odp_libconfig_ok=yes
+PKG_CHECK_MODULES([LIBCONFIG], [libconfig], [],
+		  [odp_libconfig_ok=no])
+
+if test "x$odp_libconfig_ok" = "xyes" ; then
+##########################################################################
+# Create a header file odp_libconfig_config.h which containins null
+# terminated hex dump of odp-linux.conf
+##########################################################################
+	[pushd ${srcdir}/config
+	 xxd -i odp-linux.conf | sed 's/\([0-9a-f]\)$/\0, 0x00/' > \
+	   ../platform/${with_platform}/include/odp_libconfig_config.h
```

Comment:
Generated file should go to builddir, not to srcdir.

Reply:
builddir is not defined here. Any ideas?

Reply:
Builddir is current directory.

Reply:
I need some assistance with this.  What I have now is:

```
AC_CONFIG_COMMANDS([odp_libconfig_config.h],
[
pushd ${ac_abs_top_builddir}/config
 xxd -i odp-linux.conf | sed 's/\([0-9a-f]\)$/\0, 0x00/' > \
   ../platform/${with_platform}/include/odp_libconfig_config.h
 popd], 
 [with_platform=$with_platform]
 )
```

And in .../linux-generic/Makefile.am:
```
include/odp_libconfig_config.h: $(top_srcdir)/config/odp-linux.conf $(top_srcdir)/config.status
	./$(top_srcdir)/config.status odp_libconfig_config.h
```
This seems to work as it detects changes in odp-linux.conf, but I'm not sure if this is the proper way to do this.

Reply:
I'd say
```m4
AC_CONFIG_COMMANDS([odp_libconfig_config.h],
[ mkdir -p platform/${with_platform}/include
 xxd -i ${srcdir}/platform/${with_platform}/odp-linux.conf | sed 's/\([0-9a-f]\)$/\0, 0x00/' > \
   platform/${with_platform}/include/odp_libconfig_config.h])
```

Reply:
Makefile part:
```make
include/odp_libconfig_config.h: $(top_srcdir)/config/odp-linux.conf $(top_srcdir)/config.status
	cd $(top_builddir) && $(SHELL) ./config.status odp_libconfig_config.h
```

Also it might be nice to use platform/${with_platform}/odp_libconfig_config.h as tag name, so that tags for different platforms won't collide.

Reply:
Typical idiom is
```make
include/odp_libconfig_config.h: $(top_srcdir)/config/odp-linux.conf $(top_builddir)/config.status
cd $(top_builddir) && $(SHELL) ./config.status $(subdir)/$@
```

I'd suggest you to try building with separate builddir, it should help you to catch build/src dir differences.

Reply:
Thanks! I'll try this.